core/src/main/kotlin/com/tailoredapps/countriesexample/core/remote/RemoteModule.kt

```diff
@@ -16,38 +16,36 @@
 
 package com.tailoredapps.countriesexample.core.remote
 
-import com.google.gson.Gson
+import com.jakewharton.retrofit2.converter.kotlinx.serialization.asConverterFactory
 import com.tailoredapps.androidutil.network.networkresponse.NetworkResponseRxJava2CallAdapterFactory
 import com.tailoredapps.countriesexample.core.BuildConfig
 import io.reactivex.schedulers.Schedulers
+import kotlinx.serialization.json.Json
+import okhttp3.MediaType
 import okhttp3.OkHttpClient
 import okhttp3.logging.HttpLoggingInterceptor
 import org.koin.dsl.module
 import retrofit2.Retrofit
 import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
-import retrofit2.converter.gson.GsonConverterFactory
 
 internal val remoteModule = module {
-    single { provideOkHttpClient() }
-    single { provideApi<CountriesApi>(okHttpClient = get(), gson = get(), baseUrl = get()) }
+    factory { HttpLoggingInterceptor().apply { level = HttpLoggingInterceptor.Level.BODY } }
```

Comment:
I would like to discuss the change to provide the HttpLoggingInterceptor as a factory / in the koin ecosystem..

On the one hand, it's nice to provide all instances using Koin, and not create them within the providers of another instance.
But if I don't want to change the interceptors properties, I would say it would be enough to use one instance for multiple clients, in which I would see this as a `single {  }`.
For the use-case, that someone wants to change the level, explicitely setting it here would be not needed.
And, for the use-case that for another okhttp instance the logger should be different (e.g. a timber log with a custom tag, which ofte is usefull to differentiate picasso and retrofit requests, e.g.), a new instance woul need to be created anyways, as the logger instance is final and can only be provided in the constructor.

But I'm still not sure if these arguments are enough to not use the koin ecosystem for the logging-interceptor..